### PR TITLE
Align file.edit child surfacing and docs

### LINKED ISSUE #433: [Follow-up] Align file.edit surfacing, docs, and tracker state
# [Follow-up] Align `file.edit` surfacing, docs, and tracker state

**Labels**: `enhancement`, `tools-plugins`
**Depends on**: existing `file.edit` implementation on `dev`
**Scope**: product-surface alignment, not core runtime implementation

## Problem

`file.edit` already exists in the runtime on `dev`, but surrounding product surfaces still lag behind the implementation.

Current gaps include:

1. Delegate/subagent child tool defaults still allow only `file.read` and `file.write`
2. Some docs and architecture references still describe file tooling as read/write only
3. Open tracker issues still represent `file.edit` as missing work

This creates confusion for users and maintainers: the capability exists, but some defaults, docs, and planning artifacts still imply it does not.

### 1. Delegate / subagent surfacing
- Review whether `file.edit` should be included in the default child tool allowlist
- If yes, update the default config and any related tests
- If no, document the reason explicitly

### 2. Docs alignment
Update user-facing and architecture-facing references that still describe file tools as read/write only.

### 3. Tracker cleanup
- Update or close stale checklist items in `#220` / `#292`
- Ensure future planning documents do not describe `file.edit` as missing core functionality

## Acceptance Criteria

- `file.edit` status is accurate across current docs and issue trackers
- Delegate/subagent policy for `file.edit` is explicit and tested
- No active tracker item still describes `file.edit` as an unimplemented core tool unless it refers to a clearly narrower follow-up scope

## Out of Scope

- Re-implementing the core `file.edit` runtime
- Designing a new edit API
- `apply_patch` or other separate editing tools

### PR DESCRIPTION
**Pull request by gh-xj**
## Summary

- Problem: `file.edit` already exists in the runtime, but the default child/delegate tool surface and several docs still lagged behind that reality.
- Why it matters: delegates/subagents were still defaulting to a read/write-only file surface, and the stale docs/issues made the product state look inconsistent.
- What changed: added `file.edit` to the default delegate child tool allowlist and updated the directly coupled session/subagent fixtures and tests; aligned architecture, roadmap, security, and harness docs with the existing runtime; clarified that filesystem tools use filesystem capability checks plus file-policy/path-sandbox enforcement and audit rather than the full `PolicyEngine` rule path.
- What did not change (scope boundary): no new `file.edit` runtime implementation, no change to the parent tool surface, and no broader redesign of tool policy or approval semantics.

## Linked Issues

- Closes #433
- Related #422, #292

## Change Type

- [ ] Bug fix
- [x] Feature
- [ ] Refactor
- [x] Documentation
- [x] Security hardening
- [ ] CI / workflow / release

## Touched Areas

- [ ] Kernel / policy / approvals
- [ ] Contracts / protocol / spec
- [ ] Daemon / CLI / install
- [ ] Providers / routing
- [x] Tools
- [ ] Browser automation
- [ ] Channels / integrations
- [x] ACP / conversation / session runtime
- [ ] Memory / context assembly
- [x] Config / migration / onboarding
- [x] Docs / contributor workflow
- [ ] CI / release / workflows

## Risk Track

- [ ] Track A (routine / low-risk)
- [x] Track B (higher-risk / policy-impacting)

If Track B, fill these in:

- Risk notes: this changes the default child/delegate tool allowlist, so subagents can now call `file.edit` without an explicit per-run override.
- Rollout / guardrails: the change is limited to the default child tool surface; existing filesystem policy/path sandboxing still governs access, and the coupled fixtures/tests were updated to lock the default contract.
- Rollback path: remove `file.edit` from the default child allowlist and revert the coupled fixture/doc updates.

## Validation

- [x] `cargo fmt --all -- --check`
- [x] `cargo clippy --workspace --all-targets --all-features -- -D warnings`
- [x] `cargo test --workspace --locked`
- [x] `cargo test --workspace --all-features --locked`
- [x] Relevant architecture / dep-graph / docs checks for touched areas
- [ ] Additional scenario, benchmark, or manual checks when behavior changed
- [ ] If this changes config/env fallback, limits, or defaults: include before/after behavior and regression coverage for explicit path, fallback path, and boundary values
- [ ] If tests mutate process-global env: document how state is restored or serialized

Commands and evidence:

```text
cargo fmt --check
cargo clippy --workspace --all-targets --all-features -- -D warnings

task verify
- passed locally in /Users/xj/github/loongclaw/loongclaw/.worktrees/issue433-file-edit-surfacing-cleanup
- included workspace fmt/clippy/tests, docs checks, architecture/dep-graph/conventions/harness checks, GitHub labels checks, and cargo deny
```

## User-visible / Operator-visible Changes

- Delegates/subagents now include `file.edit` in the default child tool allowlist.
- Current docs now describe `file.edit` as part of the shipped file-tool surface.

## Failure Recovery

- Fast rollback or disable path: revert the default allowlist change in `crates/app/src/config/tools.rs` and the coupled fixture/doc updates.
- Observable failure symptoms reviewers should watch for: child sessions unexpectedly gaining file mutation capability where read/write-only behavior was assumed, or further doc drift around which policy path governs filesystem tools.

## Reviewer Focus

- Confirm the default child-tool expansion in `crates/app/src/config/tools.rs` matches the intended product policy.
- Check that the session/subagent fixture updates cover the default-path contract without masking opt-in/opt-out behavior.
- Verify the security wording is now consistent between `docs/SECURITY.md` and `docs/design-docs/harness-engineering.md`.

## Summary by CodeRabbit

* **New Features**
  * Added `file.edit` as a new built-in tool capability for file operations.

* **Documentation**
  * Updated architecture, security, and design documentation to reflect the new file editing tool.
  * Updated roadmap to include the delivered file editing capability.

### CODE REVIEW

File: docs/SECURITY.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ CapabilityToken → PolicyEngine → PolicyExtensionChain → Execution → Audi
 
 **Current coverage:**
 - `shell.exec` — Kernel-mediated tool execution with capability checks, shell policy extensions, and audit events
-- `file.read` / `file.write` — Kernel-mediated tool execution with filesystem capabilities, file policy extension checks, and audit events
+- `file.read` / `file.write` / `file.edit` — Kernel-mediated tool execution with filesystem capabilities, file policy extension checks, and audit events
```
[the remark this answers is not in the document]
Comment (gh-xj) [reply docs/SECURITY.md]: Clarified in `607d90c`: `docs/SECURITY.md` now distinguishes the full Rule-of-Two `PolicyEngine` path used by `shell.exec` from the current filesystem-tool path used by `file.read`, `file.write`, and `file.edit` (filesystem capability checks plus file-policy/path-sandbox enforcement and audit, without the full `PolicyEngine` rule path). This matches the current TD-002 description in `docs/design-docs/harness-engineering.md`.

**Review by chumyin — APPROVED** [review]
hi @gh-xj, thanks for pushing this through. i pulled it into a clean worktree, corrected the security wording drift so the docs match the actual kernel policy chain, and reran fmt, clippy, workspace tests, and all-features tests locally. the `file.edit` surfacing change and the coupled fixture updates look good now. congrats on getting this one over the line.
